wfcGen.py

```diff
@@ -222,22 +229,22 @@ def collapse(self):
 
 
     def getFinalImage(self, saveFileName="result.png"):
-        if self.GIMAGE is None:
-            self.GIMAGE = Image.new("RGB", (self.sx*self.tileSize, self.sy*self.tileSize))
+        self.GIMAGE = Image.new("RGB", (self.sx*self.tileSize, self.sy*self.tileSize))
         for y in range(self.sy):
             for x in range(self.sx):
-                if self.grid[y][x].id != -1:
+                if self.grid[y][x].collapsed:
                     self.GIMAGE.paste(self.cells[self.grid[y][x].id]["img"], (x*self.tileSize, y*self.tileSize, x*self.tileSize + self.tileSize, y*self.tileSize + self.tileSize))
         self.GIMAGE.save(saveFileName)
-
+    
+    def imagePixelsToArray(self, colorToPixelDict):
+        return [[colorToPixelDict[self.GIMAGE.getpixel((x, y))] if self.GIMAGE.getpixel((x, y)) in colorToPixelDict.keys() else -1 for x in range(self.sx*self.tileSize)] for y in range(self.sy*self.tileSize)]

```

Comment:
I'm pretty sure there is a more readable way to write this but ok. As long as this is a very short function targeting a very narrow purpose with the good function name and parameter name, I'm gonna accept it.